ntrans_gui.py

```diff
@@ -0,0 +1,260 @@
+
+        ngram_check_label.grid(sticky="N", column=0, row=1, padx=(20, 0), pady=(10, 10))
+
+        select_all_checkbutton.grid(sticky="W", column=0, row=2, padx=(40, 0), pady=(0, 10))
+
+        for n in n_to_ngram_checkbox:
+            n_to_ngram_checkbox[n].grid(sticky="W", column=0, padx=(40, 0), pady=(0, 0))
+
+        data_size_label.grid(sticky="W", column=0, row=8, padx=(20, 0), pady=(30, 0))
+        ngram_data_size.grid(sticky="W", column=0, row=9, padx=(20, 0), pady=(5, 0))
+
+        target_language_label.grid(sticky="W", column=0, row=10, padx=(20, 20), pady=(30, 0))
+        target_language.grid(sticky="W", column=0, row=11, padx=(20, 20), pady=(0, 0))
+
+        filepath_label.grid(sticky="W", column=0, row=12, padx=(20, 0), pady=(30, 10))
+        show_filepath.grid(sticky="W", column=0, row=13, padx=(20, 0), pady=(0, 10))
+        get_savepath_button.grid(sticky="W", column=1, row=13, padx=(0, 20), pady=(0, 10))
+
+        generate_dictionary.grid(column=0, row=14, columnspan=2, padx=(0, 0), pady=(30, 30))
+
+        about_help_buttonframe.grid(sticky="E", column=0, row=15, columnspan=2, padx=(20, 20), pady=(0, 10))
+        about_button.pack(side="left")
+        help_button.pack(side="right")
+
+        # Black turn on formatting
+        # fmt: on
+
+    def get_save_file_path(self) -> None:
+        self.savepath: str = filedialog.askdirectory()  # type: ignore
+        if self.savepath:
+            self.filepath.set(self.savepath)
+
+    def generate_ntrans_dictionary(self) -> None:
+        self.user_choices = {
+            "save_path": self.filepath.get(),
+            "included_ngrams": [
+                n for n, var in self.checkbox_vars.items() if var.get()
+            ],
+            "amount_of_ngrams": self.data_size_var.get(),
+            "target_language": self.target_language_var.get(),
+        }
+
+        for key in self.user_choices:
+            if not self.user_choices[key]:
+                print(
+                    "All fields need to be filled in before generating the N-Trans dictionary."
+                )  # TODO: Change print to label
+                break
+
+        # TODO: Check filepath for validity
+
+        # TODO: Call logic Here
+
+    def open_help_page(self) -> None:
+        webbrowser.open_new_tab(
+            "https://www.google.com"
+        )  # TODO: Write help document and link to it
+
+    def select_all_ngrams(self) -> None:
+        if self.select_all_var.get():
+            for n in self.checkbox_vars:
+                self.checkbox_vars[n].set(True)
+        else:
+            for n in self.checkbox_vars:
+                self.checkbox_vars[n].set(False)
+
+    def open_about_window(self) -> None:
+        self.about_window = AboutWindow()
+
+    def control_path_validity(self) -> None:
+        pass
+
+    def update_ngram_checkbox(self, n: int) -> None:
+        if self.checkbox_vars[n].get():
+            self.checkbox_vars[n].set(True)
```

Comment:
This is not necessary: if it is true then set it to true.

Reply:
> First time I did a review on phone. Not at home now because our dog is having issues

Poor doggo :(